meta-iotconnect/recipes-apps/temp/iot-connect_0.1.bb

```diff
@@ -0,0 +1,81 @@
+#based on https://www.digikey.co.uk/en/maker/projects/intro-to-embedded-linux-part-6-add-custom-application-to-yocto-build/509191cec6c2418d88fd374f93ea5dda
+SUMMARY = "Compile and install the basic-sample from the IoT Connect C SDK"
+DESCRIPTION = "this recipe pulls the C SDK from the IoT Connect git repos; Sets up the basic-sample as the target to be built using CMake and then installed into the image."
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+inherit cmake
+
+DEPENDS += " curl"
+DEPENDS += " pkgconfig"
+DEPENDS += " openssl"
+DEPENDS += " util-linux-libuuid"
+
+# Where to keep downloaded source files (in tmp/work/...)
+C="${WORKDIR}/git"
+S="${C}/samples/basic-sample"
+
+SRCREV_FORMAT="machine_meta"
+
+# Where to find source files (can be local, GitHub, etc.)
+SRC_URI = "git://github.com/avnet-iotconnect/iotc-generic-c-sdk.git;\
```

Comment:
Can't we just do gitsm on the main repo and pull all the submodules? Setting the commits indivialually for each of the git modules is coumbersome to maintain. The project does not depen on those repos bewlo - the generic SDK does.

If possible to refer to a label instead of a commit, I think that we can apply a nice label to the generic C sdk and make it not have the commit number.

Reply:
hopefully fixed in this commit: 06d96649db7a951915269ac4b9d1278a992157f7

Reply:
It seems like you misunderstood what I asked.

I meant to use gitrsm on this line:
```
SRC_URI = "git://github.com/avnet-iotconnect/iotc-generic-c-sdk.git;
```
with the intent of completely getting rid of the other lines which pull explicitly cJSON, iotc-c-lib etc. I'm no expert on bitbake recipes, but I'd assume should make it pull its own dependencies that area already in gitmodules of the iotc-generic-c-sdk.git in the locations you awant them to be.

If you NEED to pull the individual submodules (like cJSON), then it sounds to me like you absolutely should stick to a fixed revision - latest revision should not be pulled every time someone runs the recipe, as the latest version of the depenedincy may no longer be compatible with the generic SDK. 

If you were to keep those lines for cJSON etc, if you deccide one day to pull the latest generic SDK, I think you would have to cross-check every depeneciny explicity defined in your recipe and make sure it is up to date with whatever revision is in the repo. That's why I thought that making for this change would save you the trouble.